doc/reference.rst

```diff
@@ -150,15 +150,30 @@ extends
 
 .. _extends-cache-buildout-option:
 
-
 extends-cache
-  An optional directory to cache remote configurations in.  This
-  allows remote configurations to be used in :ref:`offline mode
-  <offline-mode>`.  When not in offline mode, and a remote
-  configuration is used, by using a URL in an
-  :ref:`extends option <extends_option>`, the configuration is written
-  to the extends cache and will be used if the buildout is later run
-  in offline mode.
+  An optional directory to cache remote configurations in.  Remote
+  configuration is configuration specified using a URL in an
+  :ref:`extends option <extends_option>` or as the argument to the
+  :ref:`-C buildout command-line option <-C-option>`. How the
+  extends-cache behaves depends on the buildout mode:
+
+  +---------------------------------+------------------------------+
+  | Mode                            | Behavior                     |
+  +=================================+==============================+
+  | :ref:`install-from-cache        | Configuration is retrieved   |
+  | <install-from-cache-mode>` or   | from cache if possible. If   |
+  | :ref:`offline <offline-mode>`   | configuration isn't cached,  |
+  |                                 | the buildout fails.          |
+  +---------------------------------+------------------------------+
+  | :ref:`non-newest                | Configuration is retrieved   |
+  | <non-newest-mode>`              | from cache if possible. If   |
+  |                                 | configuration isn't cached,  |
+  |                                 | then it is downloaded        |
+  |                                 | and saved in the cache.      |
+  +---------------------------------+------------------------------+
+  | Default                         | Configuration is downloaded  |
```

Comment:
I'd say `Configuration is always downloaded and saved in the cache. If there's anything in the cache, it's overwritten.`, just to be perfectly clear.